### PR TITLE
New InsuranceDataForm for fillForm step

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

New, mostly independent implementation of edit step

We're only reusing `priceIntentAtoms` and `AutomaticField` from previous implementation, other things are either copied (`FormGridNew`) or newly written

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4468** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/PriceCalculatorPage/FormGridNew.css.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { style, createVar } from '@vanilla-extract/css'
+import { tokens } from 'ui'
+
+export const grid = style({
+  display: 'grid',
+  gridTemplateColumns: 'repeat(6, 1fr)',
+  gap: tokens.space.xs,
```
Comment (alebedev) [inline apps/store/src/components/PriceCalculatorPage/FormGridNew.css.ts:7]: Old version had xxs gap which looks too small

File: apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    return autoFormat(section.preview.fieldName, value)
+  }, [section, translateLabel, autoFormat])
+
+  const handleEdit = () => setActiveSectionId(section.id)
+
+  return (
+    <div
+      className={clsx(xStack({}), sprinkles({ padding: 'md' }))}
```
Comment (guilhermespopolin) [inline apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx:35]: We can pass sprinkles properties to `xStack`/`yStack`

```suggestion
      className={xStack({ padding: 'md' })}
```

File: apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      style={{ backgroundColor: tokens.colors.backgroundStandard, borderRadius: tokens.radius.md }}
```
Comment (guilhermespopolin) [inline apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx:36]: Maybe create a class for it? One that could include `xStack` on it's implementation:

`export const myClass = style([xStack({padding: 'md' }), { backgroundColor: tokens.colors.backgroundStandard }])`

**Review by guilhermespopolin — APPROVED** [review]
Noice. Just left some small comments.

**Comment by alebedev** [comment]
### Merge activity

* **Jul 11, 4:14 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4468) rebased this pull request as part of a merge.

File: apps/store/src/components/PriceCalculatorPage/useConfirmPriceIntent.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { currentPriceIntentIdAtom } from '@/components/ProductPage/PurchaseForm/priceIntentAtoms'
+import { BankSigneringEvent } from '@/services/bankSignering'
+import {
+  ExternalInsuranceCancellationOption,
+  usePriceIntentConfirmMutation,
+} from '@/services/graphql/generated'
+
+type ConfirmPriceIntentOptions = {
+  onSuccess: () => void
+  onError: (message: string) => void
```
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/components/PriceCalculatorPage/useConfirmPriceIntent.tsx:14]: Depends on whether forgetting to handle error is definite programmer error or not. I'll figure it out when implementing error handling

File: apps/store/src/components/PriceCalculatorPage/useConfirmPriceIntent.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+}
+export const useConfirmPriceIntent = ({ onSuccess, onError }: ConfirmPriceIntentOptions) => {
+  const priceIntentId = useAtomValue(currentPriceIntentIdAtom)
+  const { t } = useTranslation('purchase-form')
+  const [confirmPriceIntent] = usePriceIntentConfirmMutation({
+    variables: { priceIntentId: priceIntentId! },
```
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/components/PriceCalculatorPage/useConfirmPriceIntent.tsx:20]: There's an edge case when priceIntent is empty during initial loading. I'll try to get rid of it, but for now this is the workaround

File: apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx (hunk 3 above)
Comment (alebedev) [reply apps/store/src/components/PriceCalculatorPage/SectionPreview.tsx:36]: Yes, probably. I'm mostly ignoring structure of styling code for now, it'll be another pass once all behavior is finished

**Pull request merged**